python/tskit/tables.py

```diff
@@ -493,12 +498,15 @@ def _text_header_and_rows(self, limit=None):
             else:
                 md = base64.b64encode(metadata[j]).decode("utf8")
                 location_str = ",".join(map(str, location[j]))
+                parents_str = ",".join(map(str, parents[j]))
                 rows.append(
-                    "{}\t{}\t{}\t{}".format(j, flags[j], location_str, md).split("\t")
+                    "{}\t{}\t{}\t{}\t{}".format(
+                        j, flags[j], location_str, parents_str, md
+                    ).split("\t")
                 )
         return headers, rows
 
-    def add_row(self, flags=0, location=None, metadata=None):
+    def add_row(self, flags=0, location=None, parents=None, metadata=None):
```

Comment:
For strict compatibility we should  put the ``parents`` at the of the args list. I'm tempted to make everything after flags a keyword-only argument though. @benjeffery, any thoughts?

Reply:
Same point applies in a few other places where we're updating the public API.

Reply:
A few months back we considered reviewing the API to make more args keyword-only. I think it's a good idea.

Reply:
I also think this is a good idea. Much of this commit is just duplicating the `location` table code. I feel like a lot of this can be streamlined.